tests/functional/testMultihop.js

```diff
@@ -53,590 +39,595 @@ describe('Server list', function() {
+            await vpn.waitForQuery(countryId.visible());
+
+            await vpn.setQueryProperty( 
+                queries.screenHome.serverListView.COUNTRY_VIEW, 'contentY',
+                parseInt(await vpn.getQueryProperty(countryId, 'y')));
+
+            await vpn.wait();
+
+            if (currentCountryCode === server.code) {
+                assert(
+                    await vpn.getQueryProperty(countryId, 'cityListVisible') ===
+                    'true');
+            }
+
+            if (await vpn.getQueryProperty(countryId, 'cityListVisible') ===
+                'false') {
+                await vpn.clickOnQuery(countryId);
+            }
+
+            for (let city of server.cities) {
+                const cityId = queries.screenHome.serverListView.generateCityId(
+                    countryId, city.name);
+                console.log('  Waiting for cityId:', cityId);
+                await vpn.waitForQuery(cityId.visible().prop(
+                    'checked',
+                    currentCountryCode === server.code &&
+                        currentCity === city.localizedName));
+            }
+        }
+    })
+
+     
+    it('Pick cities for entries', async () => {
+        await vpn.waitForQueryAndClick(
+            queries.screenHome.serverListView.MULTIHOP_SELECTOR_TAB.visible());
+        await vpn.waitForQueryAndClick(
+            queries.screenHome.serverListView.ENTRY_SERVER_BUTTON.visible());
+
+        for (let server of servers) {
+            const countryId = queries.screenHome.serverListView.generateCountryId(server.code);
+
+            await vpn.waitForQuery(countryId.visible());            
+            await vpn.scrollToQuery(
+                queries.screenHome.serverListView.COUNTRY_VIEW, countryId);
+
+            if (await vpn.getQueryProperty(countryId, 'cityListVisible') === 'false') {
+                await vpn.waitForQueryAndClick(countryId.visible());
+            }
+
+            await vpn.waitForQuery(countryId.prop('cityListVisible', true));
+
+            for (let city of server.cities) {
+                console.log('  Start test for city:', city);
+                const cityId = queries.screenHome.serverListView.generateCityId(
+                    countryId, city.name);
+                await vpn.waitForQuery(cityId.visible());
+
+                await vpn.setQueryProperty(
+                    queries.screenHome.serverListView.COUNTRY_VIEW, 'contentY',
+                    parseInt(await vpn.getQueryProperty(cityId, 'y')) +
+                        parseInt(await vpn.getQueryProperty(countryId, 'y')));
+                await vpn.waitForQuery(cityId.visible());
+                
+                await vpn.getQueryProperty(cityId, 'radioButtonLabelText');
+
+                await vpn.waitForQueryAndClick(cityId.visible());
+                await vpn.waitForQuery(queries.screenHome.STACKVIEW.ready());
+                await vpn.wait(1000)
```

Comment:
This should not needed. Maybe you want to check the state of some stackview or loader obj.